pkg/cloud/vsphere/context/machine_role.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package context
+
+import (
+	"regexp"
+
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+)
+
+// MachineRole is the role of a Machine
+type MachineRole string
+
+const (
+	// ControlPlaneRole indicates a machine is a member of the control plane.
+	ControlPlaneRole MachineRole = "controlplane"
+
+	// NodeRole indicates a machine is a member of the cluster.
+	NodeRole MachineRole = "node"
+)
+
+// GetMachineRole returns the Machine's role.
+func GetMachineRole(machine *clusterv1.Machine) MachineRole {
```

Comment:
Why deviate from the way CAPI determines which machine is a control plane machine or not?
There is a utils method `IsControlPlaneMachine` in CAPI that gives that answer.

[CAPI method: IsControlPlaneMachine](https://github.com/kubernetes-sigs/cluster-api/blob/91179deda81cc4b96740a1d182f64a293844dae5/pkg/util/util.go#L134-L137)
```// IsControlPlaneMachine checks machine is a control plane node.
func IsControlPlaneMachine(machine *clusterv1.Machine) bool {
	return machine.Spec.Versions.ControlPlane != ""
}
```

Reply:
Hi @sidharthsurana,

Please see this [Slack conversation](https://kubernetes.slack.com/archives/C8TSNPY4T/p1560807049390100) where I dug into this very same issue the other day :)

Reply:
Resolved in 1009d0afb6e811276a645302f719c85e8ad67acc